urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/RandomPermutationClass.java

```diff
@@ -0,0 +1,183 @@
+package permlib.examples;
+
+import java.util.ArrayList;
+import java.util.Random;
+import permlib.PermUtilities;
+import permlib.Permutation;
+import permlib.Permutations;
+import permlib.classes.PermutationClass;
+
+/**
+ * Investigate building a class randomly by choosing basis elements of
+ * increasing length from within the class.
+ *
+ * @author Michael Albert
+ */
+public class RandomPermutationClass {
+
+    public static final Random R = new Random();
+
+    public static void main(String[] args) {
+        
+        rand321(12,5);
+        
+//        for (int i = 0; i < 200; i++) {
+//            example01(5);
+            //System.out.println();
+        // }
+//      // example02();
+
+//        for (Permutation p : ul(6)) {
+//            System.out.println(p);
+//        }
+//        System.out.println();
+//        for (Permutation p : ur(6)) {
+//            System.out.println(p);
+//        }
+//        System.out.println();
+//        for (Permutation p : ul(7)) {
+//            System.out.println(p);
+//        }
+//        System.out.println();
+//        for (Permutation p : ur(7)) {
+//            System.out.println(p);
+//        }
+//        System.out.println();
+
+//        for(Permutation p : io(7)) System.out.println(p);
+    }
+
+    /* In Av(321) randomly choose a new basis element of each length */
+    public static void example01(int k) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.